### PR TITLE
System channel

### PR DESCRIPTION
**Pull request by craftxbox**
## Purpose
This PR adds structure for unhidable/restricted channels, and adds one by default: The System channel.

## Technical implementation notes (Optional if exclusively visual changes)
Gives channels a new 'Unhidable' boolean, which if set makes the channel not possible to be specifically filtered to, or filtered out. 
Adds a list of 'Restricted' Channels (by name) in ChatRegister That players cannot speak in ( is this the right way?)
Adds a new SYSTEM channel that is Unhidable and listed in the Restricted channels list by default.

first pr! please let me know if i've done something horribly stupid.

### CODE REVIEW

File: Assets/Scripts/Chat/ChatChannel.cs [hunk 1]
```diff
@@ -7,4 +7,5 @@ public struct ChatChannel
     public string Name;
     public string Abbreviation;
     public Color Color;
+    public bool Unhidable;
```
Comment (Dyhr) [inline Assets/Scripts/Chat/ChatChannel.cs]: Negatives can sometimes cause confusing, for example in `if (!Unhidable)`. Prefer `Hidable`.

File: Assets/Scripts/Chat/ChatChannels.cs [hunk 2]
```diff
@@ -12,4 +12,12 @@ public List<ChatChannel> GetChannels()
...
         return Channels;
     }
+    public List<ChatChannel> GetUnhidable()
+    {
+        return Channels.FindAll(channel => {return channel.Unhidable;});
+    }
+    public List<ChatChannel> GetHidable()
+    {
+        return Channels.FindAll(channel => {return !channel.Unhidable;});
+    }
```
Comment (Dyhr) [inline Assets/Scripts/Chat/ChatChannels.cs]: I am not 100% sure on this, but won't an `IEnumerator<ChatChannel>` cause less garbage allocations?

Comment (Dyhr) [reply Assets/Scripts/Chat/ChatChannels.cs]: Resolved in voice chat

**Comment by Em3rgencyLT** [comment]
Looks good! Congrats on first PR!

**Pull request merged**